### PR TITLE
Eventbrite Bug Fixes

### PR DESCRIPTION
**Pull request by nateh777**
##### What does the change add or fix?

- Fixed an issue with our Eventbrite Person field type on the group member list by making it a proper filter now.
- Fixed an issue with Ticket quantities not being counted properly when there were multiple orders for the same person.

---------

##### What does the change add or fix in a succinct statement that will be read by clients?

- Fixed an issue with our Eventbrite Person field type on the group member list by making it a proper filter now.
- Fixed an issue with Ticket quantities not being counted properly when there were multiple orders for the same person.

---------

##### Who reported, requested, or paid for the change?

Warranty

---------

##### Does this update or add options to the block UI?

image

image

---------

##### What files does it affect?

- rocks.kfs.EventBrite/Eventbrite.cs
- rocks.kfs.EventBrite/Field/Types/EventbritePersonFieldType.cs

---------

##### Is it a breaking change for other versions/clients?

No, though we recommend re-syncing your groups if you want valid ticket quantities on each group member.

### CODE REVIEW

File: rocks.kfs.EventBrite/Eventbrite.cs [hunk 1]
```diff
@@ -268,7 +269,6 @@ public static void SyncOrder( string apiUrl )
                 var personAliasService = new PersonAliasService( rockContext );
 
                 AttendanceOccurrence occ = GetOrAddOccurrence( rockContext, group, ebEvent );
```
Comment (nateh777) [inline rocks.kfs.EventBrite/Eventbrite.cs:271]:
```suggestion
                AttendanceOccurrence occ = GetOrAddOccurrence( rockContext, group, ebEvent );

```

File: rocks.kfs.EventBrite/Eventbrite.cs [hunk 2]
```diff
@@ -346,7 +352,8 @@ private static void SyncAttendee( RockContext rockContext, Attendee attendee, Or
 
                     groupMemberService.Add( member );
 
-                    SetPersonData( rockContext, person, attendee, group, IsRSVPEvent && order.Attendees != null ? order.Attendees.Count : attendee.Quantity, EnableLogging );
```
Comment (nateh777) [inline rocks.kfs.EventBrite/Eventbrite.cs:349]: Need to test Quantity only event.

File: rocks.kfs.EventBrite/Eventbrite.cs [hunk 3]
```diff
@@ -313,8 +313,14 @@ public static void SyncAttendee( string apiUrl )
 
         }
 
-        private static void SyncAttendee( RockContext rockContext, Attendee attendee, Order order, Group group, GroupMemberService groupMemberService, PersonAliasService personAliasService, AttendanceOccurrence occ, long evntid, bool IsRSVPEvent, string gmPersonAttributeKey, bool updatePrimaryEmail, int recordStatusId = 5, int connectionStatusId = 66, bool EnableLogging = false )
+        private static void SyncAttendee( RockContext rockContext, Attendee attendee, Order order, Group group, GroupMemberService groupMemberService, PersonAliasService personAliasService, AttendanceOccurrence occ, long evntid, bool IsRSVPEvent, string gmPersonAttributeKey, bool updatePrimaryEmail, int recordStatusId = 5, int connectionStatusId = 66, bool EnableLogging = false, List<string> groupMembersAdded = null )
         {
+            if ( groupMembersAdded == null )
```
Comment (nateh777) [inline rocks.kfs.EventBrite/Eventbrite.cs:318]:
```suggestion
            // If the groupMembersAdded list comes in as null we need to be able to identify it did, as in the case of a webhook call with an individual order. 
            if ( groupMembersAdded == null )
```

**Review by guymaness2 — APPROVED** [review]

File: rocks.kfs.EventBrite/Eventbrite.cs (hunk 2 above)
Comment (nateh777) [reply rocks.kfs.EventBrite/Eventbrite.cs:349]: @guymaness2 tested Quantity Only event. It seems to work just fine as expected.

**Comment by nateh777** [comment]
@guymaness2 ready to merge

**Pull request merged**